test/gateway/SlotDataReader.sol

```diff
@@ -3,15 +3,16 @@ pragma solidity ^0.8.0;
 
 import {GatewayFetchTarget, IGatewayVerifier} from '../../contracts/GatewayFetchTarget.sol';
 import {GatewayFetcher, GatewayRequest} from '../../contracts/GatewayFetcher.sol';
+import {Ownable} from '@openzeppelin/contracts/access/Ownable.sol';
 
-contract SlotDataReader is GatewayFetchTarget {
+contract SlotDataReader is GatewayFetchTarget, Ownable {
     using GatewayFetcher for GatewayRequest;
 
     IGatewayVerifier public _verifier;
     address public _target;
     address public _pointer;
 
-    constructor(IGatewayVerifier verifier, address target) {
+    constructor(IGatewayVerifier verifier, address target) Ownable(msg.sender) {
```

Comment:
Why?

Reply:
To set `TrustVerifier` confirmation, the caller must be the contract or the owner of the contract.

https://github.com/unruggable-labs/unruggable-gateways/blob/0a11cc0c719c455a969500b3249ebacdba8e4e60/test/gateway/common.ts#L222-L232
Before the `SlotDataReader` test is run, the `TrustedVerifier` gets configured by the `admin` (owner of `SlotDataReader`)